source/Halibut.Tests/Support/BackwardsCompatibility/ProxyHalibutTestBinaryRunner.cs

```diff
@@ -184,9 +192,9 @@ public class RoundTripRunningOldHalibutBinary : IDisposable
 
             public void Dispose()
             {
-                compatBinaryStayAlive.Dispose();
                 cts.Cancel();
-                runningOldHalibutTask.GetAwaiter().GetResult();
```

Comment:
Or is it that we don't consume another thread waiting for the task to complete? Should this be `DisposeAsync`?

Reply:
Cancellation first I think wins and then there is no need to wait on that task.

It can still hang but not as frequently! I think there may be an issue with the way the binary is started and all the tasks that creates as I can cause the test to hang eventually by preventing the manual reset event from being signaled